### PR TITLE
Update documentation for GameLevel and GameLevelDefaultImpl

### PR DESCRIPTION
**Pull request by JoelTroch**
##### Opendev - Team n°3
-----

This pull request update the documentation of GameLevel and GameLevelDefaultImpl for better understanding of their purposes.

### CODE REVIEW

File: src/main/java/gameframework/game/GameLevel.java [hunk 1]
```diff
@@ -1,7 +1,14 @@
 package gameframework.game;
 
+/**
+ * Interface for game levels.
```
Comment (cdlm) [inline src/main/java/gameframework/game/GameLevel.java]: Useless, this paraphrases the Java declaration.

File: src/main/java/gameframework/game/GameLevel.java [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+ *
+ * Technically speaking, a "GameLevel" is just a thread whose initialize/simulate/free the level.
```
Comment (cdlm) [inline src/main/java/gameframework/game/GameLevel.java]: Please double-check the grammar

- /technically speaking/: says nothing
- /is just a thread/: …so? How is it used in the rest of the framework? Why is a thread needed?

File: src/main/java/gameframework/game/GameLevel.java [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
+ */
 public interface GameLevel extends Runnable {
+	/** Start the level (aka thread). */
```
Comment (cdlm) [inline src/main/java/gameframework/game/GameLevel.java]: Can you explain when/in which conditions this can/should be used, and what will be the result?

**Review by cdlm — CHANGES_REQUESTED** [review]
Comments that paraphrase the code are not helpful; needs more details to be worth merging.

**Comment by JoelTroch** [comment]
I applied the suggestions above, let us know what you think.

**Comment by cdlm** [comment]
👍

**Pull request closed without merging**